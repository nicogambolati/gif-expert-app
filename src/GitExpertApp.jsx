import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GitExpertApp = () => {
  const [categories, setCategories] = useState(["Dragon Ball Z"]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GitExpertApp</h1>

      <AddCategory onNewCategory={(event) => onAddCategory(event)} />

      {categories.map((category) => {
        return <GifGrid key={category} category={category} />;
      })}
    </>
  );
};
