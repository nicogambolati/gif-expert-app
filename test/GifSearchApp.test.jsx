import { render, screen } from "@testing-library/react";
import { GifSearchApp } from "../src/GifSearchApp";

describe("Test in <GifSearchApp />", () => {
  test("should match snapshot", () => {
    const { container } = render(<GifSearchApp />);
    expect(container).toMatchSnapshot();
    // screen.debug();
  });
});
