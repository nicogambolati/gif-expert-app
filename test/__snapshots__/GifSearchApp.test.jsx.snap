// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifSearchApp /> should match snapshot 1`] = `
<div>
  <h1>
    Gif Search App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="test"
      value=""
    />
  </form>
  <h3>
    Dragon Ball Z
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
